Armadillo/src/main/java/com/scribd/armadillo/extensions/SharedPrefExt.kt

```diff
@@ -0,0 +1,85 @@
+package com.scribd.armadillo.extensions
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties.BLOCK_MODE_GCM
+import android.security.keystore.KeyProperties.ENCRYPTION_PADDING_NONE
+import android.security.keystore.KeyProperties.PURPOSE_DECRYPT
+import android.security.keystore.KeyProperties.PURPOSE_ENCRYPT
+import android.util.Log
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
+import com.scribd.armadillo.Constants.DI.STANDARD_STORE_FILENAME
+import com.scribd.armadillo.Constants.Keys.ANDROID_KEYSTORE_NAME
+import java.io.File
+import java.security.KeyStore
+
+fun SharedPreferences.deleteSharedPreference(context: Context, filename: String, keystoreAlias: String) {
```

Comment:
This should be named specific to the encryption prefs, since it also does things with the keystore.